temporal-sdk/src/main/java/io/temporal/activity/Activity.java

```diff
@@ -217,39 +45,14 @@ public static ActivityExecutionContext getExecutionContext() {
   }
 
   /**
-   * If there is a need to return a checked exception from an activity do not add the exception to a
-   * method signature but rethrow it using this method. The library code will unwrap it
-   * automatically when propagating exception to the caller. There is no need to wrap unchecked
-   * exceptions, but it is safe to call this method on them.
-   *
-   * <p>The reason for such design is that returning originally thrown exception from a remote call
-   * (which child workflow and activity invocations are ) would not allow adding context information
-   * about a failure, like activity and child workflow id. So stubs always throw a subclass of
-   * {@link ActivityFailure} from calls to an activity and subclass of {@link ChildWorkflowFailure}
-   * from calls to a child workflow. The original exception is attached as a cause to these wrapper
-   * exceptions. So as exceptions are always wrapped adding checked ones to method signature causes
-   * more pain than benefit.
-   *
-   * <p>Throws original exception if e is {@link RuntimeException} or {@link Error}. Never returns.
-   * But return type is not empty to be able to use it as:
-   *
-   * <pre>
-   * try {
-   *     return someCall();
-   * } catch (Exception e) {
-   *     throw Activity.wrap(e);
-   * }
-   * </pre>
-   *
-   * If wrap returned void it wouldn't be possible to write <code>
-   * throw Activity.wrap</code> and compiler would complain about missing return.
+   * Use this to rethrow a checked exception from an Activity instead of adding the exception to a method signature.
```

Comment:
I would add this information to the documentation before removing it from the JavaDoc.

Reply:
It is there, though could be more discoverable: https://docs.temporal.io/docs/java/activities/#throwing-activity-errors